src/org/parosproxy/paros/model/SiteMap.java

```diff
@@ -745,5 +743,26 @@ private void clearParentFilter (SiteNode parent) {
 			clearParentFilter(parent.getParent());
 		}
 	}
+	
+	@Override
+	public void removeNodeFromParent(MutableTreeNode node) {
+		super.removeNodeFromParent(node);
+		if(node.getParent() == getRoot()) {
+			ZAP.getEventBus().publishSyncEvent(SiteMapEventPublisher.getPublisher(), 
+	        		new Event(SiteMapEventPublisher.getPublisher(), SiteMapEventPublisher.SITE_REMOVED_EVENT, new Target((SiteNode)node)));

```

Comment:
This can be extracted to a method (one that takes the SiteNode and event type), it will reduce  code duplication a little further.